### PR TITLE
Massachusetts 2023 Child and Dependent Tax Credit changes

### LINKED ISSUE #3436: Massachusetts 2023 Child and Dependent Tax Credit changes
The amounts per dependent have increased to $310 in 2023, and will further increase to $440 from 2024 onwards. The bill also removed the two-children cap.

https://malegislature.gov/Bills/193/H4104

### PR DESCRIPTION
**Pull request by jiajiagong0123**
Fixes #3436

### Summary
📝📈🗑️

This pull request updates the `policyengine_us` package to incorporate the recent legislative changes to the Massachusetts Child and Dependent Tax Credit for 2023 and 2024. It modifies the relevant parameters and tests in the `gov/states/ma/tax/income/credits` subdirectory and adds the corresponding sources and changelog entry.

> _The MA dependent credit changed_
> _To help families rearrange_
> _Their tax burden and plan_
> _For `amount` and `cap` span_
> _In `ma_dependent_credit` and tests arranged_

### Walkthrough
* Bump minor version and add changelog entry for MA dependent credit changes ([link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L1-R3))
* Add new values and remove cap for MA dependent credit amount for 2023 and 2024 ([link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-c5ffd70f95e657c71f7289d45f1af3b558fc07838dcf73c4cf44e92663e97371R4-R5), [link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-c5ffd70f95e657c71f7289d45f1af3b558fc07838dcf73c4cf44e92663e97371R12-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-9a8cbd8a3be179910abcb9c631540cf8c7a368d17a8582c1c33d640a067d97c4R4), [link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-9a8cbd8a3be179910abcb9c631540cf8c7a368d17a8582c1c33d640a067d97c4R11-R14))
* Update and add test cases for MA dependent credit calculation for 2022, 2023, and 2024 ([link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-1f287c6dd6ed1484d4be210b69d20f5a03f2da041a46291e41d013ccd65f5821L1-R1), [link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-1f287c6dd6ed1484d4be210b69d20f5a03f2da041a46291e41d013ccd65f5821L22-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/3446/files?diff=unified&w=0#diff-1f287c6dd6ed1484d4be210b69d20f5a03f2da041a46291e41d013ccd65f5821L46-R136))

### CODE REVIEW

**Review by MaxGhenis — COMMENTED** [review]
Can you add a legal code reference?

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+  2024-01-01: 440
 metadata:
   reference:
     - title: Form 1 2021 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2021-form-1-instructions/download#page=17
     - title: Form 1 2022 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2022-form-1-instructions/download#page=17
+    - title: Draft Form 1 2023 Massachusetts Resident Income Tax (line 46)
+      href: https://www.mass.gov/doc/draft-2023-form-1-instructions/download#page=17
+    - title: Governor Healey Signs First Tax Cuts in More Than 20 Years
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:14]: dont need this reference

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 2]
```diff
@@ -1,12 +1,20 @@
 description: MA dependent credit amount.
 values:
   2021-01-01: 180
+  2023-01-01: 310
+  2024-01-01: 440
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:5]: delete for now and file an issue for 2024 child and dependent tax credit changes and include this reference:
https://www.mass.gov/news/governor-healey-signs-first-tax-cuts-in-more-than-20-years

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
   reference:
     - title: Form 1 2021 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2021-form-1-instructions/download#page=17
     - title: Form 1 2022 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2022-form-1-instructions/download#page=17
+    - title: Draft Form 1 2023 Massachusetts Resident Income Tax (line 46)
+      href: https://www.mass.gov/doc/draft-2023-form-1-instructions/download#page=17
+    - title: Governor Healey Signs First Tax Cuts in More Than 20 Years
+      href: https://www.mass.gov/news/governor-healey-signs-first-tax-cuts-in-more-than-20-years
+    - title: Legal Code Bill S. 1959 Section 4 (y)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:16]:
```suggestion
    - title: Legal Code Bill S. 1959 Section 4 (y) (iii) (B)
```

File: changelog_entry.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Massachusetts 2023 Child and Dependent Tax Credit changes.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Massachusetts 2023 child and dependent tax credit rules.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Thanks for adding the bill but please add a legal code reference as well. Also run `make format`

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+  2023-01-01: 310
 metadata:
   reference:
     - title: Form 1 2021 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2021-form-1-instructions/download#page=17
     - title: Form 1 2022 Massachusetts Resident Income Tax (line 46)
       href: https://www.mass.gov/doc/2022-form-1-instructions/download#page=17
+    - title: Draft Form 1 2023 Massachusetts Resident Income Tax (line 46)
+      href: https://www.mass.gov/doc/draft-2023-form-1-instructions/download#page=17
+    - title: Law	Result Summary Part I Title IX Chapter 62 Section 6 (y) (ii) (B)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:13]: Please do the same for the cap. And doesn't the bill change the age limit from under 12 to under 13? In any case, please add the legal reference to the age limit while we're here.
```suggestion
    # Does not yet reflect 2023 changes as of 2023-12-14.
    - title: Massachusetts General Laws Part I Title IX Chapter 62 Section 6 (y) (ii) (B)
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 6]
```diff
@@ -1,12 +1,19 @@
 description: MA dependent credit amount.
 values:
   2021-01-01: 180
+  2023-01-01: 310
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:4]: This isn't in the bill you cited, so please cite this one instead which Governor Healey signed: https://malegislature.gov/Bills/193/H4104

Note the 2024 change occurs in Section 22, and Section 45 indicates that it takes effect in 2024.
```suggestion
  2023-01-01: 310
  2024-01-01: 440
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/dependent_cap.yaml [hunk 7]
```diff
@@ -1,12 +1,19 @@
 description: Maximum dependents for MA dependent credit.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/dependent_cap.yaml:1]:
```suggestion
description: Massachusetts limits its child and dependent tax credit to this number of dependents.
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/dependent_cap.yaml [hunk 8]
```diff
@@ -1,12 +1,19 @@
...
       href: https://www.mass.gov/doc/2022-form-1-instructions/download#page=17
+    - title: Draft Form 1 2023 Massachusetts Resident Income Tax (line 46)
+      href: https://www.mass.gov/doc/draft-2023-form-1-instructions/download#page=17
+    - title: Law	Result Summary Part I Title IX Chapter 62 Section 6 (y) (ii) (B)
+      href: https://malegislature.gov/Laws/GeneralLaws/PartI/TitleIX/Chapter62/Section6 
+    - title: Bill S. 1959 Section 4 (y) (iii) (B)  
+      href: https://malegislature.gov/Bills/193/S1959  
   unit: people
   period: year
   label: MA dependent credit dependent cap
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/dependent_cap.yaml:19]:
```suggestion
  label: Massachusetts child and dependent credit dependent cap
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 9]
```diff
@@ -1,12 +1,19 @@
 description: MA dependent credit amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:1]:
```suggestion
description: Massachusetts provides this amount for each dependent in its child and dependent credit.
```

File: policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml [hunk 10]
```diff
@@ -1,12 +1,19 @@
...
       href: https://www.mass.gov/doc/2022-form-1-instructions/download#page=17
+    - title: Draft Form 1 2023 Massachusetts Resident Income Tax (line 46)
+      href: https://www.mass.gov/doc/draft-2023-form-1-instructions/download#page=17
+    - title: Law	Result Summary Part I Title IX Chapter 62 Section 6 (y) (ii) (B)
+      href: https://malegislature.gov/Laws/GeneralLaws/PartI/TitleIX/Chapter62/Section6     
+    - title: Bill S. 1959 Section 4 (y) (iii) (B)  
+      href: https://malegislature.gov/Bills/193/S1959    
   unit: currency-USD
   period: year
   label: MA dependent credit
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ma/tax/income/credits/dependent/amount.yaml:19]:
```suggestion
  label: Massachusetts child and dependent credit amount
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Also for more info see @keepjojoyoung's [blog post draft](https://docs.google.com/document/d/1Tbkyyb3Ei0APDUcNNWiUWgvVlY68lNiFdR1LN31g_a0/edit)

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**